partner_address_format_alternative/models/res_country.py

```diff
@@ -0,0 +1,15 @@
+# Copyright 2025 Quartile (https://www.quartile.co)
+# License AGPL-3.0 or later (https://www.gnu.org/licenses/agpl).
+
+from odoo import fields, models
+
+
+class ResCountry(models.Model):
+    _inherit = "res.country"
+
+    address_format_alt = fields.Text(string="Alternative Address Format")
+    address_format_alt_domain = fields.Char(
+        string="Domain for Alternative Address Format",
+        help="Python domain condition to use the alternative address format on partners. "
+        "Example: [('company_id.country_id.code', '!=', 'JP')]",
+    )
```

Comment:
This will unfortunately not work in many cases as the partner may not have a company assigned. How about just adding a text field `address_format_domestic` and use its field value if set for domestic transactions instead?